optuna/integration/tensorflow.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import absolute_import
+
+import optuna
+
+
+try:
+    import tensorflow as tf
+    from tensorflow.train import SessionRunHook
+    _available = True
+except ImportError as e:
+    _import_error = e
+    # TensorFlowPruningHook is disabled because TensorFlow is not available.
+    _available = False
+    SessionRunHook = object
+
+
+class TensorFlowPruningHook(SessionRunHook):
+    """TensorFlow SessionRunHook to prune unpromising trials.
+
+    Example:
+
+        Add a pruning SessionRunHook for a TensorFlow's Estimator.
+
+        .. code::
+
+                pruning_hook = TensorFlowPruningHook(
+                    trial=trial,
+                    estimator=clf,
+                    metric="accuracy",
+                    is_higher_better=True,
+                    run_every_steps=10,
+                )
+                hooks = [pruning_hook]
+                tf.estimator.train_and_evaluate(
+                    clf,
+                    tf.estimator.TrainSpec(input_fn=train_input_fn, max_steps=500, hooks=hooks),
+                    eval_spec
+                )
+    Args:
+        trial:
+            A :class:`~optuna.trial.Trial` corresponding to the current evaluation of
+            the objective function.
+        estimator:
+            An estimator which you will use.
+        metric:
+            An evaluation metric for pruning, e.g., ``accuracy`` and ``loss``.
+        is_higher_better:
+           It should be True if you use a metric to be maximize such as ``loss``.
+        run_every_steps:
+           An interval to watch the summary file.
+    """
+    def __init__(self, trial, estimator, metric, is_higher_better, run_every_steps):
+        # type: (optuna.trial.Trial, tf.estimator.Estimator, str, bool, int) -> None
+
+        self.trial = trial
+        self.estimator = estimator
+        self.current_summary_step = -1
+        self.metric = metric
+        self.is_higher_better = is_higher_better
+        self._global_step_tensor = None
+        self._timer = tf.train.SecondOrStepTimer(every_secs=None, every_steps=run_every_steps)
+
+    def begin(self):
+        # type: () -> None
+
+        self._global_step_tensor = tf.train.get_global_step()
+
+    def before_run(self, run_context):
+        # type: (tf.train.SessionRunContext) -> tf.train.SessionRunArgs
+
+        del run_context
+        return tf.train.SessionRunArgs(self._global_step_tensor)
+
+    def after_run(self, run_context, run_values):
+        # type: (tf.train.SessionRunContext, tf.train.SessionRunValues) -> None
+
+        global_step = run_values.results
+        # Get eval metrics every n steps
```

Comment:
Please add trailing periods to here and [L81](https://github.com/pfnet/optuna/pull/292/files#diff-6777d6b27690da66a18f118b10e4a1dcR87), because it is an Optuna's coding convention.

Reply:
I fixed them in L78 and L87.
You mentioned to L81, but is it L87?

Reply:
Oops..., sorry, L87 is correct.